### PR TITLE
Async to sync

### LINKED ISSUE #271: Counterproductive use of async routes in auth?
Hi, I was just browsing through this repo and noticed that the auth routes are declared using `async def`, in https://github.com/rednafi/fastapi-nano/blob/master/app/core/auth.py#L125 but the underlying database access in the function that is also used as a dependency on line https://github.com/rednafi/fastapi-nano/blob/master/app/core/auth.py#L117 is synchronous. My understanding of FastAPI is such that in this case, it would be better to declare the route and dependency using a simple `def`, so that the whole thing is handled in a thread and other requests can still be processed concurrently. The way it's currently implemented, I think the entire event loop might be blocked whenever the database is accessed in that route and the related dependency. 

Just flagging out of interest, apologies if my understanding is incorrect here.

### PR DESCRIPTION
**Pull request by kareemmahlees**
### Changes
- changes async functions to sync 
- fixed some typing issues

The tests needed no change and they all still run fine.
The change was not massive but I am looking forward to contributing more in the future 🙂 

closes #271

### CODE REVIEW

File: app/core/auth.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ async def get_current_user(token: str = Depends(oauth2_scheme)) -> UserInDB:
 
 
 @router.post("/token", response_model=Token)
-async def login_for_access_token(
```
Comment (rednafi) [inline app/core/auth.py:125]: Following the convention in the views, we probably want to keep this `async`.

File: app/core/auth.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def get_password_hash(password: str) -> str:
 
 
 def get_user(
-    db: dict[str, dict[str, str]],
+    db: dict[str, dict[str, Any]],
```
Comment (rednafi) [inline app/core/auth.py:56]: Good catch!

**Comment by kareemmahlees** [comment]
@rednafi Any comments? if all is fine there is still one pending workflow that needs to be run or approved idk :)

**Comment by rednafi** [comment]
> @rednafi Any comments? if all is fine there is still one pending workflow that needs to be run or approved idk :)

Check the review comment please. It looks good, other than one minor issue.

**Comment by rednafi** [comment]
> > @rednafi Any comments? if all is fine there is still one pending workflow that needs to be run or approved idk :)
> 
> @kareemmahlees Check the review comment please. It looks good, other than one minor issue.

Merged with the fix, thank you!

**Pull request merged**